apps/native/app/src/screens/notifications/notifications.tsx

```diff
@@ -62,11 +79,52 @@ export const NotificationsScreen: NavigationFunctionComponent = ({
   const { data, loading, error, fetchMore } = useGetUserNotificationsQuery({
     variables: { input: { limit: DEFAULT_PAGE_SIZE } },
   })
+  const showError = error && !data
 
   const [markUserNotificationAsRead] = useMarkUserNotificationAsReadMutation()
   const [markAllUserNotificationsAsSeen] =
     useMarkAllNotificationsAsSeenMutation()
 
+  const [markAllUserNotificationsAsRead] =
+    useMarkAllNotificationsAsReadMutation({
+      onCompleted: (data) => {
+        if (data.markAllNotificationsRead?.success) {
+          // If all notifications are marked as read, update cache to reflect that
+          client.cache.modify({
```

Comment:
@eirikurn I tried to get it to work with `optimisticResponse` and/or the real `data` but no luck (time-boxed it). We can maybe take a look at it next week if we want to try more ☺️